packages/backend/src/Util.ts

```diff
@@ -44,4 +46,91 @@ export async function makeID<Uid>(hasCollision: Function, length=6){ // default
   // TODO: this causes the request to fail, but it also crashes
   if(i == maxIter) throw new InternalServerError("Failed to generate new ID");
   return id;
+}
+
+interface ImageKitLayer{
+  [key: string]: string | number
+}
+const formatImageKitURL = (layers: ImageKitLayer[]) : string => {
+  return [
+    'https://ik.imagekit.io/equalvote/tr:',
+    ...layers.map(
+      l => [
+        l['type'],
+        ...[Object.entries(l).filter(([k, v]) => k != 'type').map(([k, v]) => `${k}-${encodeURIComponent(v)}`)],
+        ((l['type'] as string).startsWith('l')? 'l-end' : '')
+      ].join(',')
+    ).join(':'),
+    '/media_preview.png'
+  ].join('')
+}
+
+interface TagObject{
+  [key: string]: string
+}
+let ElectionsModel =  ServiceLocator.electionsDb();
+export async function getMetaTags(req: any) : Promise<TagObject>  {
+  let parts = req.url.split('/');
+  let election:Election|null;
+
+  // list copied from App.tsx
+  if(['', 'About', 'ElectionInvitations', 'ElectionsYouManage', 'ElectionsYouVotedIn', 'OpenElections', 'Sandbox'].includes(parts[1].split('?')[0])){
+    election = null;
+  }else{
+    const electionID = (parts[1] == 'Election' ? parts[2] : parts[1])
+    election = await ElectionsModel.getElectionByID(electionID, req);
+  }
+
+  let len = election?.races[0]?.candidates.length ?? 0;
+  let n_hidden = (len > 5) ? 1 : (5-len);
+  let n_cropped = (len > 5) ? 0 : (5-len);
+
+  return {
+      __META_TITLE__: election?.title ?? 'dev.star.vote',
+      __META_DESCRIPTION__: election?.description ?? "Create secure elections with voting methods that don't spoil the vote.",
+      __META_IMAGE__: election == null ?
+        'https://assets.nationbuilder.com/unifiedprimary/pages/1470/attachments/original/1702692040/Screenshot_2023-12-15_at_6.00.24_PM.png?1702692040' :
+        // https://imagekit.io/docs/transformations#position-of-layer
+        // https://imagekit.io/docs/add-overlays-on-images#list-of-supported-text-transformations-in-text-layers
+        formatImageKitURL([
+          // Draw the election title
+          {
+            type: 'l-text',
+            i: election?.title,
+            w: 900,
+            lx: 500-(900/2),
+            ly: 180,
+            fs: 45,
+            ff: 'Montserrat-Black.ttf',
+          },
+          // Hide the unused rows
+          {
+            type: 'l-image',
+            i: 'ik_canvas',
+            bg: 'FFFFFF',
+            lfo: 'bottom_left',
+            w: 1000,
+            h: 22+n_hidden*110+((n_hidden%2 == 0) ? -10 : 10),
+          },
+          // Draw the candidates from the first race
+          ...election.races[0].candidates.splice(0,5).map((c, i) => ({
+            type: 'l-text',
+            i: (i == 4 && (election?.races[0]?.candidates.length ?? 0) > 5)? `+${(election?.races[0]?.candidates.length ?? 0) - 4} more` : c.candidate_name,
```

Comment:
Might want to limit the length of the candidate names here, unless imagekit does something to prevent the names from going over the bubbles.

Reply:
Good point, I added a fix. Imagekit will automatically wrap to multiple lines, but I added truncation on top of that to help make sure it doesn't become 3 lines

![image](https://github.com/user-attachments/assets/b4c6498a-3da8-4a39-b0a6-a27b789499a9)
![image](https://github.com/user-attachments/assets/5092656a-ebb9-4d7e-9526-5723f2639652)